gentoopm/pkgcorepm/pkg.py

```diff
@@ -103,6 +111,14 @@ def _iter_maints():
         return tuple.__new__(self, _iter_maints())
 
 
+class PkgCoreUpstream(PMUpstream):
+    def __getattribute__(self, key: str) -> typing.Any:
+        if key == "remote_ids":
+            return super().__getattribute__(key)
+        raise NotImplementedError("This metadata attribute is not implemented "
+                                  "for pkgcore")
```

Comment:
While I get what you're doing here, it's awful for typing. Please don't. Just make defaults in `PMUpstream` raise it, I guess, since you'll need to override them anyway, right?